src/library/analysis/analysisfeature.cpp

```diff
@@ -160,6 +160,15 @@ void AnalysisFeature::analyzeTracks(const QList<AnalyzerScheduledTrack>& tracks)
                 &TrackAnalysisScheduler::finished,
                 this,
                 &AnalysisFeature::onTrackAnalysisSchedulerFinished);
+        // Forward the signal to be picked up by Library.
+        // Used by WOverview to render the re-analysis progress of loaded tracks.
+        // This is the equivalent to PlayerManager's progress signal fired for
+        // analysis triggered by loading a track.
+        connect(m_pTrackAnalysisScheduler.get(),
+                &TrackAnalysisScheduler::trackProgress,
+                this,
+                &AnalysisFeature::trackProgress,
+                Qt::DirectConnection);
```

Comment:
We should not use a direct connection here, this may cause headaches later. See my other comment.